doc/apiary/v2/fiware-ngsi-simple-cookbook.apib

```diff
@@ -0,0 +1,872 @@
++ Parameters
+    + limit: 20 (optional, number) - Limit the number of entities to be retrieved
+    + offset: 20 (optional, number) - Skip a number of records. Pagination
+
++ Response 200 (application/json)
+
+        [
+         {
+            "type": "Room",
+            "id": "DC_S1_D41",
+            "temperature": 35.6
+         },
+         {
+            "type": "Room",
+            "id": "Boe_Idearium",
+            "temperature": 22.5
+         },
+         {
+            "type": "Car",
+            "id": "P-9873-K",
+            "speed": {
+                "value": 100,
+                "type": "number",
+                "accuracy": 2,
+                "timestamp": {
+                    "value": "2015-06-04T07:20:27.378Z",
+                    "type": "date"
+                }
+            }
+         }
+        ]
+
+### List entities by id [GET /v2/entities{?id}]
+
++ Parameters
+    + id: DC_S1_D41 (optional, string) - Only retrieve entities with this id
+
++ Response 200 (application/json)
+
+        [
+         {
+            "type": "Room",
+            "id": "DC_S1_D41",
+            "temperature": 35.6
+         }
+        ]
+
+### List entities by list of ids [GET /v2/entities{?id}]
+
++ Parameters
+    + id: DC_S1_D41, Boe_Idearium (optional, string) - Id list with the entities to be retrieved
+
++ Response 200 (application/json)
+
+        [
+         {
+            "type": "Room",
+            "id": "DC_S1_D41",
+            "temperature": 35.6
+         },
+         {
+            "type": "Room",
+            "id": "Boe_Idearium",
+            "temperature": 22.5
+         }
+        ]
+
+### List all entities of an specific type [GET /v2/entities{?type}]
+
++ Parameters
+    + type: Room (optional, string) - Only retrieve entities of this type
+
++ Response 200 (application/json)
+
+        [
+         {
+            "type": "Room",
+            "id": "DC_S1_D41",
+            "temperature": 35.6
+         },
+         {
+            "type": "Room",
+            "id": "Boe_Idearium",
+            "temperature": 22.5
+         }
+        ]
+
+### List all entities by id pattern matching [GET /v2/entities{?idPattern}]
+
++ Parameters
+    + pattern: DC_* (optional, string) - Only retrieve entities which id matches this pattern.
+
++ Response 200 (application/json)
+
+        [
+         {
+            "type": "Room",
+            "id": "DC_S1_D41",
+            "temperature": 35.6
+         }
+        ]
+
+### List all Entities. Pagination [GET /v2/entities{?limit,offset,count}]
+
++ Parameters
+    + count: true
+    + limit: 2
+    + offset: 2
+
++ Response 200 (application/json)
+
+    + Headers
+
+            X-Total-Count: 30
+            Link: http://context.example.org/v2/entities?offset=4&limit=2; rel="next", http://context.example.org/v2/entities?offset=28&limit=2; rel="last", http://context.example.org/v2/entities?offset=0&limit=2; rel="first", http://context.example.org/v2/entities?offset=2&limit=2; rel="prev"
```

Comment:
If "last" have to be included always, then a count() operation is always needed at DB level in addition to the operation for the query itself. This have an impact on performance (in my opinion the point of having `options=count` was to avoid unnecesary performance lost, doing the count() only in that case).